src/gui/advancedsettings.cpp

```diff
@@ -508,10 +511,19 @@ void AdvancedSettings::loadAdvancedSettings()
     addRow(DISK_IO_TYPE, tr("Disk IO type (requires restart)") + u' ' + makeLink(u"https://www.libtorrent.org/single-page-ref.html#default-disk-io-constructor", u"(?)")
            , &m_comboBoxDiskIOType);
 #endif
-    // Enable OS cache
-    m_checkBoxOsCache.setChecked(session->useOSCache());
-    addRow(OS_CACHE, (tr("Enable OS cache") + u' ' + makeLink(u"https://www.libtorrent.org/reference-Settings.html#disk_io_write_mode", u"(?)"))
-            , &m_checkBoxOsCache);
+    // Disk IO read mode
+    m_comboBoxDiskIOReadMode.addItem(tr("Disable OS Cache"), QVariant::fromValue(BitTorrent::DiskIOReadMode::DisableOSCache));
+    m_comboBoxDiskIOReadMode.addItem(tr("Enable OS Cache"), QVariant::fromValue(BitTorrent::DiskIOReadMode::EnableOSCache));
+    m_comboBoxDiskIOReadMode.setCurrentIndex(m_comboBoxDiskIOReadMode.findData(QVariant::fromValue(session->diskIOReadMode())));
+    addRow(DISK_IO_READ_MODE, (tr("Disk IO read mode") + u' ' + makeLink(u"https://www.libtorrent.org/reference-Settings.html#disk_io_read_mode", u"(?)"))
+            , &m_comboBoxDiskIOReadMode);
+    // Disk IO write mode
+    m_comboBoxDiskIOWriteMode.addItem(tr("Disable OS Cache"), QVariant::fromValue(BitTorrent::DiskIOWriteMode::DisableOSCache));
+    m_comboBoxDiskIOWriteMode.addItem(tr("Enable OS Cache"), QVariant::fromValue(BitTorrent::DiskIOWriteMode::EnableOSCache));
+    m_comboBoxDiskIOWriteMode.addItem(tr("Write Through (requires libtorrent >= 2.0.6)"), QVariant::fromValue(BitTorrent::DiskIOWriteMode::WriteThrough));
```

Comment:
IMO (for GUI only) it isn't good to use a such wording, better use `#if` to enable/disable it at compile time. Otherwise linux user might complain that they upgraded the libtorrent version from 2.0.5 to 2.0.6 but the option is not working.

Reply:
then what'll be selected incase someone downgrades?
https://github.com/qbittorrent/qBittorrent/pull/17005#discussion_r873131347

Reply:
I changed the wording a bit. See if that helps.

Reply:
> Otherwise linux user might complain that they upgraded the libtorrent version from 2.0.5 to 2.0.6 but the option is not working.

Why not increase requirements up to libtorrent 2.0.6 (for 2.0 branch)?
>better use #if to enable/disable it at compile time. 

IMO, it's preferred way for GUI.
@summerqB, don't you still understand why we can't do the same in WebUI?


> then what'll be selected incase someone downgrades?
> [#17005 (comment)](https://github.com/qbittorrent/qBittorrent/pull/17005#discussion_r873131347)

The mentioned comment is about differences between the same qBittorrent built with different branches of libtorrent.
I mean if option in config is write-through but libtorrent is 1.2 (e.g. libtorrent 2.0 based build was used previously) you should explicitly replace it with one of the supported options.

Reply:
>then what'll be selected incase someone downgrades?

From previous discussion with other developers, we reached an conclusion: It is often near impossible to provide reliable behavior for downgrades so we usually won't talk about it. We would only consider upgrade scenario.

As for this specific code: qbt will detect incompatible setting and use the default value (which is `EnableOSCache`).